Sources/GRPCNIOTransportCore/Client/Connection/Connection.swift

```diff
@@ -127,9 +127,19 @@ package final class Connection: Sendable {
   /// This function returns when the connection has closed. You can observe connection events
   /// by consuming the ``events`` sequence.
   package func run() async {
-    let connectResult = await Result {
-      try await self.http2Connector.establishConnection(to: self.address)
+    func establishConnectionOrThrow() async throws(RPCError) -> HTTP2Connection {
```

Comment:
These are the right gymnastics: the other way to do this is to be explicit with the types in the `catching:` closure:

```swift
await Result { () async throws(RPCError) -> HTTP2Connection in
  // ...
}
```